### PR TITLE
Add token validation check to math programming

### PR DESCRIPTION
**Pull request by sneaxiy**
Add error check to the math programming `objective` and `constraint` as early as possible.

### CODE REVIEW

File: python/runtime/optimize/model_generation.py [hunk 1]
```diff
@@ -12,12 +12,56 @@
...
+    Raises:
+        AssertionError if any token is invalid.
+    """
+    valid_columns = [c.lower() for c in columns]
+
+    if group_by and group_by.lower() not in valid_columns:
+        raise AssertionError("GROUP BY column %s not found" % group_by)
+
+    if not tokens:
+        return
```
Comment (typhoonzero) [inline python/runtime/optimize/model_generation.py]: No tokens should raise error?

File: python/runtime/optimize/model_generation_test.py [hunk 2]
```diff
@@ -72,12 +86,17 @@ def replace_objective_token(self, objective, old, new):
 
     def replace_constraint_token(self, constraint, old, new):
         def replace_one_constraint(c):
-            c = copy.copy(c)
-            for i, token in enumerate(c["tokens"]):
+            ret = {
```
Comment (typhoonzero) [inline python/runtime/optimize/model_generation_test.py]: use `copy.deepcopy`?

File: python/runtime/optimize/model_generation_test.py [hunk 3]
```diff
@@ -18,8 +18,22 @@
...
+        return IDENTIFIER_REGEX.fullmatch(token) is not None
+
+    def test_main(self):
+        tokens = ['a', '_', 'a123', '__', '_123']
+        for t in tokens:
+            self.assertTrue(self.is_identifier(t))
+
+        tokens = ['1', '123_', '3def']
+        for t in tokens:
+            self.assertFalse(self.is_identifier(t))
```
Comment (typhoonzero) [inline python/runtime/optimize/model_generation_test.py:37]: Add test case for `assert_are_valid_tokens`?

File: python/runtime/optimize/model_generation.py (hunk 1 above)
Comment (sneaxiy) [reply python/runtime/optimize/model_generation.py]: Done.

File: python/runtime/optimize/model_generation_test.py (hunk 2 above)
Comment (sneaxiy) [reply python/runtime/optimize/model_generation_test.py]: Great idea! Done.

File: python/runtime/optimize/model_generation_test.py (hunk 3 above)
Comment (sneaxiy) [reply python/runtime/optimize/model_generation_test.py:37]: Done.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**